src/vs/base/common/comparers.ts

```diff
@@ -93,6 +371,68 @@ function comparePathComponents(one: string, other: string, caseSensitive = false
 	return one < other ? -1 : 1;
 }
 
+/** @returns `true` if the string is starts with a lowercase letter. Otherwise, `false`. */
+function isLower(string: string) {
+	const character = string.charAt(0);
+
+	if (character.toLocaleUpperCase() !== character) {
+		return true;
+	}
+
+	return false;
+}
+
+/** @returns `true` if the string starts with an uppercase letter. Otherwise, `false`. */
+function isUpper(string: string) {
+	const character = string.charAt(0);
+
+	if (character.toLocaleLowerCase() !== character) {
+		return true;
+	}
+
+	return false;
+}
+
+/**
+ * Compares the case of the provided strings - with lowercase considered less than uppercase
+ *
+ * @returns
+ * ```text
+ *   -1 if one is lowercase and other is uppercase
+ *    1 if one is uppercase and other is lowercase
+ *    0 otherwise
+ * ```
+ */
+export function compareCaseLowerFirst(one: string, other: string): number {
```

Comment:
I exported these functions so I could test them. They seem simple, but this is actually one of the main areas that I tweaked. I originally wasn't sure whether or not uppercase and lowercase should be considered greater than non-case letters or not, and whether a word that starts with a non-case character like an underscore should be non-case because of that underscore, or should be judged by the first character that has a case. 

I can remove the exports and the tests now if desired.